sockeye/lr_scheduler.py

```diff
@@ -128,7 +154,8 @@ def __call__(self, t):
         if self.lr is None:
             assert self.base_lr is not None
             self.lr = self.base_lr
-        return self.lr
+        lr = self._warmup(t) if self.warmup > 0 and t <= self.warmup else self.lr
```

Comment:
this might lead to confusing logs if we reduce self.lr in the first steps, as we will write to the log that the learning rate got reduce, while we're still ramping up.